honeybee_energy/writer.py

```diff
@@ -20,11 +21,12 @@ def face_to_idf(face):
         '\n\t%d,\t!- Number of Vertices' \
         '\n\t%s;' % (
             face.name,
-            face.properties.face_type.name,
-            en_prop.construction.name if en_prop.construction else '',
+            face.type.name,
+            en_prop.construction.name,
             face.parent.name if face.parent else 'unknown',
             face.boundary_condition.name,
-            face.boundary_condition.boundary_condition_object_idf,
+            face.boundary_condition.boundary_condition_object_idf if
```

Comment:
Now that boundary conditions are read-only, I guess we can do this.  But I still feel that there are compelling reasons to exclude this unnecessary properties from the boundary_condition objects: 

1. It is potentially misleading to developers to have a meaningless property on most boundary conditions (people may think that they are supposed to specify a special type of special "Ground" `boundary_condition_object` for a Ground boundary condition when there is no such thing).
2. Surface boundary conditions are typically the exception to the rule and it seems weird to make all of the other boundary conditions bear this extra property just to accommodate the Surface.  And I would highlight that, now that BCs are read-only, it really makes the `__init__` of all BoundaryConditions really weird since all of the Boundary Conditions would have this init argument for boundary_condition_object (that I guess is required since the Surface BC can't exist without a boundary_condition_object?)
3. I would assume that exclusion from Python objects also meas we could exclude the unnecessary property from the jsons, which can help keep the json files smaller when someone has a model with thousands of non-surface Faces.
4. The OpenStudio measure definitely doesn't need a blank string for boundary condition object.  It's really only this IDF method that benefits from it.
5. When I eventually implement all of the different types of ground boundary conditions (including KIVA, etc.), there are going to be tons of extra properties on these boundary conditions.  It's a bit inconsistent to have the `boundary_condition_object` on all Boundary Conditions but not all of these other ground-related properties.

So I would as if all of these drawbacks really worth the convenience of not having an extra "if" statement in this IDF writer, which we have already discussed is probably not going to be a common way of generating simulation files.

Reply:
5 is good enough to keep it out. You are right that in the long therm there will be more inconsistencies so this will be just normal.